### DISCUSSION TITLE
Any idea of using R inline code control chunk option "include"?

### DISCUSSION THREAD

**Discussion by JinTonique** [quarto-dev/quarto-cli#6648] [Q&A]
### Description

I would like to control if the specific chunk is included in my output file by using `ifelse()` function of R. 

Here is an example:

I defined something previously:
```{r}
apple_is_square <- TRUE
```
Then, I don't want the following chunk included in my output file since `apple_is_square <- TRUE`. I did:

```{r}
#| label: I love my round apple
#| include: `r ifelse(apple_is_square, paste("false"), paste("true"))`

tbl <- data.frame(v1 = c("a","b","c"), v2 = c(1,2,3))
tbl
```
Apparently, it does not work in this way, quarto said:
```
Warning messages:
1: In value[[3L]](cond) : Failed to parse chunk options in body:
Error in parse(text = code, keep.source = FALSE): <text>:1:26: unexpected symbol
1: alist( label: Production share
                             ^

2: In opts$engine <- unlist(strsplit(rmdChunkOpts, split = "(\\s|,)+"))[[1]] :
  Coercing LHS to a list
```

Question: Is there any way that I can choose to include the chunk according to the value of `apple_is_square`?

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

This being said, see <https://quarto.org/docs/computations/r.html#chunk-options> (at the end of the linked section).

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by cscheid** [comment]
> Question: Is there any way that I can choose to include the chunk according to the value of apple_is_square?

Not currently. The fundamental reason is that quarto needs to support both knitr and jupyter engines, and this is not a feature that's currently supported in jupyter.

**Comment by cderv** [comment]
> ```
> #| include: `r ifelse(apple_is_square, paste("false"), paste("true"))`
> ```

You can't use inline knitr R code inside a YAML option value field. That is not a syntax that will work. Reason is simple: We need to parse the YAML before any of inline evaluation happens. 

Have a look at how chunk options are written for R chunks evaluated with **knit** engines. https://quarto.org/docs/computations/r.html#chunk-options

FWIW there is a specific feature for `engine: knitr` only that is using `!expr` in a YAML field to have R expression evaluated **at yaml parsing time**

````markdown
---
title: "Reproducible Quarto Document"
format: html
---

```{r}
apple_is_square <- TRUE
```

Then, I don't want the following chunk included in my output file since `apple_is_square <- TRUE.` 

```{r}
#| label: I love my round apple
#| include: !expr isFALSE(apple_is_square)
tbl <- data.frame(v1 = c("a","b","c"), v2 = c(1,2,3))
tbl
```
````
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/a6aca1e5-afe0-46d8-bcf4-e1aa9af3ac67)

Reminder: this works only with **knitr** engine as this is a knitr feature directly (not working with `engine: jupyter`)
